x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.test.tsx

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { EditorFrame } from './editor_frame';
+import { Visualization, Datasource, DatasourcePublicAPI } from '../../types';
+import { act } from 'react-dom/test-utils';
+
+const nextTick = () => new Promise(resolve => setTimeout(resolve));
```

Comment:
Could you use timer mocks instead of having this? https://jestjs.io/docs/en/timer-mocks

Reply:
No, because the purpose of this is to wait for the Promise result of the mocked datasource (`    initialize: jest.fn(() => Promise.resolve()),`) which is asynchronous. The timer mocks only monkey patch the timer functions, not Promises. We could mock Promises by ourselves to behave synchronously, but I would rather not hide the asynchronous nature of Promises in the test as it might shadow certain conditions. I changed the implementation to `const waitForPromises = () => new Promise(resolve => setImmediate(resolve));` which does the same thing and added a comment.